app/src/main/java/org/mozilla/focus/experiments/NimbusSetup.kt

```diff
@@ -5,53 +5,44 @@
 package org.mozilla.focus.experiments
 
 import android.content.Context
-import androidx.core.net.toUri
-import kotlinx.coroutines.runBlocking
-import mozilla.components.service.nimbus.Nimbus
 import mozilla.components.service.nimbus.NimbusApi
 import mozilla.components.service.nimbus.NimbusAppInfo
-import mozilla.components.service.nimbus.NimbusDisabled
-import mozilla.components.service.nimbus.NimbusServerSettings
+import mozilla.components.service.nimbus.NimbusBuilder
 import mozilla.components.support.base.log.logger.Logger
+import org.json.JSONObject
 import org.mozilla.experiments.nimbus.NimbusInterface
-import org.mozilla.experiments.nimbus.internal.EnrolledExperiment
 import org.mozilla.experiments.nimbus.internal.NimbusException
-import org.mozilla.experiments.nimbus.joinOrTimeout
 import org.mozilla.focus.BuildConfig
 import org.mozilla.focus.GleanMetrics.NimbusExperiments
 import org.mozilla.focus.R
 import org.mozilla.focus.ext.components
 import org.mozilla.focus.ext.settings
 import org.mozilla.focus.nimbus.FocusNimbus
 
+/**
+ * The maximum amount of time the app launch will be blocked to load experiments from disk.
+ *
+ * ⚠️ This value was decided from analyzing the Focus metrics (nimbus_initial_fetch) for the ideal
+ * timeout. We should NOT change this value without collecting more metrics first.
+ */
 private const val TIME_OUT_LOADING_EXPERIMENT_FROM_DISK_MS = 200L
 
-@Suppress("TooGenericExceptionCaught")
-fun createNimbus(context: Context, url: String?): NimbusApi {
-    val errorReporter: ((String, Throwable) -> Unit) = reporter@{ message, e ->
-        Logger.error("Nimbus error: $message", e)
-
-        if (e is NimbusException && !e.isReportableError()) {
-            return@reporter
-        }
+/**
+ * Create the Nimbus singleton object for the Focus/Klar apps.
+ */
+fun createNimbus(context: Context, urlString: String?): NimbusApi {
+    val isAppFirstRun = context.settings.getAppLaunchCount() == 0
```

Comment:
I think we should avoid using the `appLaunchCount` and instead use `isFirstRun`. I'm not sure why we didn't clean up this pref that was meant for the TipManager which we removed.

```suggestion
    val isAppFirstRun = context.settings.isFirstRun
```

Reply:
Filed BZ-1802214 to remove `getAppLaunchCount`.